src/utils/prices.ts

```diff
@@ -84,7 +84,7 @@ const IMPACT_TIERS = [
 
 type WarningSeverity = 0 | 1 | 2 | 3 | 4
 export function warningSeverity(priceImpact: Percent | undefined): WarningSeverity {
-  if (!priceImpact) return 0
+  if (!priceImpact || priceImpact.lessThan(0)) return 0
```

Comment:
Would `priceImpact?.lessThan(0)` work?

Reply:
`if (priceImpact?.lessThan(0))` wouldn't return 0 when `priceImpact` is undefined, so i'd need to inverse the boolean logic or something which would be less readable imo. lmk if i misunderstood your point there.

for `undefined` .... it does seem to be a pattern throughout the codebase to allow it. i didn't write this code originally, so i don't know why this specific function accepts undefined. my only guess is that it silences some TS warnings

Reply:
i think this is a result of the hard commitment to hooks, where when data isn't available yet lots of things just pass undefined. so somewhere a hook calls this, and the hook accepts undefined